project/cmake/modules/FindCXX11.cmake

```diff
@@ -0,0 +1,18 @@
+include(TestCXXAcceptsFlag)
+
+# try to use compiler flag -std=c++11
+check_cxx_accepts_flag("-std=c++11" CXX_FLAG_CXX11)
+if(CXX_FLAG_CXX11)
+  add_options (CXX ALL_BUILDS "-std=c++11")
+  set(CXX_STD11_FLAGS "-std=c++11")
+else()
```

Comment:
gcc >=4.7.2 is known to work, 4.6 doesn't.
not sure about clang, but I guess 3.6 works
and VS2013 afaik